### PR TITLE
Added a bunch of error messages and made some exceptions return null ...

### PR DESCRIPTION
**Pull request by DeltaRays**
...  (in the java mongodb driver exceptions are thrown when something is null)

### CODE REVIEW

File: src/main/java/fr/romitou/mongosk/skript/expressions/ExprCollection.java [hunk 1]
```diff
@@ -41,9 +41,13 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
         MongoDatabase database = exprDatabase.getSingle(e);
-        return (name == null || database == null)
-                ? new MongoCollection[0]
-                : new MongoCollection[]{database.getCollection(name)};
+        try {
+            return (name == null || database == null)
+                    ? new MongoCollection[0]
+                    : new MongoCollection[]{database.getCollection(name)};
+        } catch (IllegalArgumentException ex) {
+            return null; // A MongoDB collection with that name doesn't exist!
```
Comment (Olyno) [inline src/main/java/fr/romitou/mongosk/skript/expressions/ExprCollection.java:49]: Don't return null when you can
```suggestion
            return new MongoCollection[0]; // A MongoDB collection with that name doesn't exist!
```

File: src/main/java/fr/romitou/mongosk/skript/expressions/ExprDatabase.java [hunk 2]
```diff
@@ -41,9 +41,13 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
         MongoClient client = exprClient.getSingle(e);
-        return (name == null || client == null)
-                ? new MongoDatabase[0]
-                : new MongoDatabase[]{client.getDatabase(name)};
+        try {
+            return (name == null || client == null)
+                    ? new MongoDatabase[0]
+                    : new MongoDatabase[]{client.getDatabase(name)};
+        } catch (IllegalArgumentException ex) {
+            return null; // A MongoDB database with that name doesn't exist!
```
Comment (Olyno) [inline src/main/java/fr/romitou/mongosk/skript/expressions/ExprDatabase.java:49]: Same here
```suggestion
            return new MongoDatabase[0]; // A MongoDB database with that name doesn't exist!
```

File: src/main/java/fr/romitou/mongosk/skript/expressions/ExprValue.java [hunk 3]
```diff
@@ -55,10 +55,20 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
         if (value == null || document == null)
             return new Object[0];
         if (parseMark == 1) {
-            return new Object[]{document.get(value)};
+            try {
+                return new Object[]{document.get(value)};
+            } catch (NullPointerException ex) {
+                return null; // That document value doesn't exist;
```
Comment (Olyno) [inline src/main/java/fr/romitou/mongosk/skript/expressions/ExprValue.java:61]: Same again
```suggestion
                return new Object[0]; // That document value doesn't exist;
```

File: src/main/java/fr/romitou/mongosk/skript/expressions/ExprValue.java [hunk 4]
```diff
@@ -55,10 +55,20 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
         } else {
-            List<Object> list = document.getList(value, Object.class);
-            return (list == null) ? new Object[0] : list.toArray();
+            try {
+                List<Object> list = document.getList(value, Object.class);
+                return (list == null) ? new Object[0] : list.toArray();
+
+            } catch (ClassCastException ex) {
+                Skript.error("The mongodb document value is not a list!"); // I'm not sure whether or not this should return an error
+                return null;
```
Comment (Olyno) [inline src/main/java/fr/romitou/mongosk/skript/expressions/ExprValue.java:70]: Ah sh*t, here we go again
```suggestion
                return new Object[0];
```

**Review by Olyno — CHANGES_REQUESTED** [review]
Better with better errors, be carefull to ``return null``, it's a really bad habbit.

**Comment by DeltaRays** [comment]
Also thanks Olyno, I always returned null just because mongodb was returning null, too.
When do you think I should use null?

**Comment by Olyno** [comment]
> When do you think I should use null?

Never, it's a really bad practice which can break a lot of stuff. Skript by default returns everytime ``null``, which is really bad. That's the reason why we changed that for the new parser.
When returning an array, you can simply return a ``new YourObject[0]`` instead of ``null``. When it's not an array but a single object, I would recommand to use ``Optionnals`` of Java 8

**Comment by DeltaRays** [comment]
Thanks for the info!

**Review by Romitou — APPROVED** [review]

**Comment by Romitou** [comment]
Many thanks to DeltaRays for the Pull Request and to Olyno for the code reviews! I am merging.

**Pull request merged**